### PR TITLE
Add MobileExternalBrowser to client config

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Introducing a client config property that determines if the mobile app should use the default external browser to perform SSO Authentication instead of the internal one.

Value defaults to **false**.

Note: It does not include an admin console UI.

Important: We want to include this in the latest ESR as will mostly be used by certain enterprise customers.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-60332

#### Release Note
```release-note
Added config setting NativeAppSettings -> MobileExternalBrowser that tells the mobile app to perform the SSO Authentication using the external default browser
```

### CODE REVIEW

**Review by sbishel — CHANGES_REQUESTED** [review]

File: server/public/model/config.go [hunk 1]
```diff
@@ -2764,6 +2764,7 @@ type NativeAppSettings struct {
 	AppDownloadLink        *string  `access:"site_customization,write_restrictable,cloud_restrictable"`
 	AndroidAppDownloadLink *string  `access:"site_customization,write_restrictable,cloud_restrictable"`
 	IosAppDownloadLink     *string  `access:"site_customization,write_restrictable,cloud_restrictable"`
+	MobileExternalBrowser  bool     `access:"site_customization,write_restrictable,cloud_restrictable"`
```
[the remark this answers is not in the document]
Comment (enahum) [reply server/public/model/config.go:2767]: Alright, I'll change it to a pointer and set the default as false

**Review by sbishel — APPROVED** [review]
Thanks

**Comment by enahum** [comment]
@amyblais will we cherry pick this one for the ESR ?

**Comment by amyblais** [comment]
> will we cherry pick this one for the ESR ?

Yes, that's correct.

**Review by rahimrahman — APPROVED** [review]

**Pull request merged**